web/src/components/Diagram/components/DAG/useListenToArrowKeysDown.tsx

```diff
@@ -0,0 +1,35 @@
+import {useEffect} from 'react';
+
+export function useListenToArrowKeysDown(
+  affectedSpans: string[],
+  handleNextSpan: () => void,
+  handlePrevSpan: () => void
+): void {
+  useEffect(() => {
+    function downHandler({key}: KeyboardEvent) {
```

Comment:
This is a great improvement, but I'm a little concerned about going out of React to achieve this,  what if we instead use refs for at least the ones we can?

Reply:
I think using the ref would be a performance issue. especially if the dag is really big.
that's I I have implemented outside react